src/main/java/mate/academy/springboot/datajpa/service/CategoryServiceImpl.java

```diff
@@ -0,0 +1,35 @@
+package mate.academy.springboot.datajpa.service;
+
+import mate.academy.springboot.datajpa.model.Category;
+import mate.academy.springboot.datajpa.repository.CategoryRepository;
+import org.springframework.stereotype.Service;
+
+@Service
+public class CategoryServiceImpl implements CategoryService {
+    private final CategoryRepository categoryRepository;
+
+    public CategoryServiceImpl(CategoryRepository categoryRepository) {
+        this.categoryRepository = categoryRepository;
+    }
+
+    @Override
+    public Category save(Category category) {
+        return categoryRepository.save(category);
+    }
+
+    @Override
+    public Category getById(Long id) {
+        return categoryRepository.getById(id);
+    }
+
+    @Override
+    public void delete(Long id) {
+        categoryRepository.deleteById(id);
+    }
+
+    @Override
+    public void update(Long id, Category category) {
+        category.setId(id);
+        categoryRepository.update(category.getId(), category.getName());
+    }
```

Comment:
here you should accept category with id already set
setting of id and other mapping stuff is mappers job